packages/nodes-base/nodes/Kafka/KafkaTrigger.node.ts

```diff
@@ -194,14 +201,20 @@ export class KafkaTrigger implements INodeType {
 				eachMessage: async ({ topic, message }) => {
 
 					let data: IDataObject = {};
-					let value = message.value?.toString() as string;
+					let value: string | Buffer = message.value?.toString() as string;
 
 					if (options.jsonParseMessage) {
 						try {
 							value = JSON.parse(value);
 						} catch (error) { }
 					}
 
+					if (options.useRawMessage) {
+						try {
+							value = message.value as Buffer;
```

Comment:
Maybe we should use `await this.helpers.prepareBinaryData(message.value as Buffer);` instead ?

Reply:
How do you want to use the data? what type of data are you expecting to get when you use useRawMessage?

Reply:
In my use-case, data in the kafka topic is protobuf and with the nominal behaviour (original data is stringified), when I try to deserialize the protobuf, it doesn't work. 
That's why I would like keep the raw data to use it in another node which deserialize my protobuf data.

Reply:
@d3rwan any updates on this one? Just trying to get my head around it :) It seems like a good idea to have but I am wondering if it is generic enough would there ever be a case to have a raw message that isn't a buffer or a prefered binary data object?

Reply:
@Joffcom sorry for the delay ! 

It's not easy to share an example workflow because I obviously have specific node which read encoded protobuf and parse it it JSON, and I can't share it here. 

But my workflow is like that : 
```
{ "nodes": [ { "parameters": { "topic": "identity-events", "groupId": "n8n-client-id", "options": {} }, "name": "Kafka Trigger", "type": "n8n-nodes-base.kafkaTrigger", "typeVersion": 1, "position": [ 460, 300 ], "credentials": { "kafka": { "id": "1", "name": "Kafka account" } } }, { "parameters": { "message": "message", "targetField": "identityEvent" }, "name": "Identity Event Decoder", "type": "@swan-io/n8n-nodes-swan.identityEventDecoder", "typeVersion": 1, "position": [ 680, 300 ] }, { "parameters": {}, "name": "NoOp", "type": "n8n-nodes-base.noOp", "typeVersion": 1, "position": [ 900, 300 ] } ], "connections": { "Kafka Trigger": { "main": [ [ { "node": "Identity Event Decoder", "type": "main", "index": 0 } ] ] }, "Identity Event Decoder": { "main": [ [ { "node": "NoOp", "type": "main", "index": 0 } ] ] } } }

```

The node `Identity Event Decoder` will transform a protobuf buffer (I use https://github.com/stephenh/ts-proto for encoding/decoding) in a JS object. 

And with the default behaviour, I think the toString method can alter buffer sometimes, and then the decoding no longer works

Reply:
Hey @d3rwan,

So at the moment this change would potentially only be useful for someone who has the extra node you have? It seems like a good feature to add but testing is possibly tricky :)

I will have a chat with the team internally to see what they want to do with this one.